sleap_nn/training/model_trainer.py

```
@@ -813,554 +848,905 @@ def train(self):
-                    edges = [
-                        (x.source.name, x.destination.name)
-                        for x in self.skeletons[0].edges
-                    ]
-                    head_config[key]["edges"] = edges
+        if not Path(self.dir_path).exists():
+            try:
+                Path(self.dir_path).mkdir(parents=True, exist_ok=True)
+            except OSError as e:
+                message = f"Cannot create a new folder in {self.dir_path}. Check the permissions to the given Checkpoint directory. \n {e}"
+                logger.error(message)
+                raise OSError(message)
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)